### PR TITLE
fix: unity catalog import from write_deltalake

### PR DESCRIPTION
**Pull request by jaychia**
Closes: #3629

### CODE REVIEW

File: daft/dataframe/dataframe.py [hunk 1]
```diff
@@ -874,7 +874,12 @@ def write_deltalake(
         from daft.io import DataCatalogTable
         from daft.io._deltalake import large_dtypes_kwargs
         from daft.io.object_store_options import io_config_to_storage_options
-        from daft.unity_catalog import UnityCatalogTable
+
+        _UNITY_AVAILABLE = True
+        try:
+            from daft.unity_catalog import UnityCatalogTable
```
Comment (desmondcheongzx) [inline daft/dataframe/dataframe.py:880]: We can do `from daft.dependencies import unity_catalog`

File: daft/dataframe/dataframe.py [hunk 2]
```diff
@@ -884,30 +889,35 @@ def write_deltalake(
...
+            storage_options = table._storage_options or {}
+            new_storage_options = io_config_to_storage_options(io_config, table_uri)
+            storage_options.update(new_storage_options or {})
+        else:
             if isinstance(table, str):
                 table_uri = table
             elif isinstance(table, pathlib.Path):
                 table_uri = str(table)
-            elif isinstance(table, UnityCatalogTable):
+            elif _UNITY_AVAILABLE and isinstance(table, UnityCatalogTable):
```
Comment (desmondcheongzx) [inline daft/dataframe/dataframe.py:906]: Which then let's us do
```
elif unity_catalog.module_available() and isinstance(table, unity_catalog.UnityCatalogTable):
```

Comment (jaychia) [reply daft/dataframe/dataframe.py]: Thanks, this is great and works!

**Review by desmondcheongzx — APPROVED** [review]
Looks good, thanks for the fast fix!

**Comment by kevinzwang** [comment]
Merging this in

**Pull request merged**